dao/src/main/java/se/fortnox/reactivewizard/db/DbProxy.java

```diff
@@ -119,7 +121,9 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                 databaseConfig);
             statementFactories.put(method, observableStatementFactory);
         }
-        return observableStatementFactory.create(args, connectionProvider);
+        Observable<Object> resultObservable = observableStatementFactory.create(args, connectionProvider);
```

Comment:
Doh! I guess I need a revisit on this one. It is kind of similar to the wrappers in the client. Perhaps we could have a generic wrapper type to avoid creating multiple wrappers. Will see if that works.

Reply:
This is now solved, but will be refactored in the future.